### ISSUE TITLE
Add an alarm mechanism if the agent runs out of funds

### ISSUE THREAD

**Issue by fredo** [beamer-bridge/beamer#788]
We need an alarm if agent runs out of funds. Easy way would be to do it via prometheus .

**Comment by istankovic** [comment]
- spin up a new thread to detect when an agent is low on ETH or on any of the configured tokens
- introduce thresholds for tokens and ETH (low amounts) in the config file
- make a script that e-mails us when the testnet agents are low on funds
-

**Comment by andfletcher** [comment]
I have some questions that are not clear to me.

Will the agents run this service on their own? Will they be able to run a setup/initiation script that will let them define the parameters, e.g., what email address the warning should be sent to? What is the threshold, how many times should the person receive an email per day / what are the check intervals once he runs out of funds? Is it just once or should it be done multiple times?

**Comment by fredo** [comment]
Is this done by the monitoring script? @compojoom

**Comment by istankovic** [comment]
> Is this done by the monitoring script? @compojoom

I think the idea here was to provide this feature as part of the agent itself (see https://github.com/beamer-bridge/beamer/issues/788#issuecomment-1200963272).

**Comment by fredo** [comment]
Yes, but since we have the monitoring script, which is supposed to run aside to the agent (possibly adding it to the docker compose), wouldn't this be a better fit?

**Comment by fredo** [comment]
Note, that the monitoring script is planned to support chat updates and direct connection to the owner of the agent anyways.

**Comment by istankovic** [comment]
> Note, that the monitoring script is planned to support chat updates and direct connection to the owner of the agent anyways.

Ah, in that case, yes, it makes sense to delegate all of that to the monitoring script.
Liquidity providers should not rely on us in order to monitor their agents. :+1:
